### DISCUSSION TITLE
Wireshark Formula

### DISCUSSION THREAD

**Discussion by wilkmaciej** [Homebrew/discussions#1452] [Everyday usage] [answered]
Why wireshark formula doesn't build wireshark binary and Wireshark.app?
Wireshark doesn't provide dmg file with built wireshark.app for now due to some issues with automated builds if I'm not mistaken [Wireshark GitLab issue](https://gitlab.com/wireshark/wireshark/-/issues/17294) so `brew install --cask wireshark` works for now only for Intel based machines.
Is there an option to resolve this issue and build wireshark with homebrew Formula?

**Comment by SMillerDev** [comment]
It does build the binary, `.app` files just generally don't belong in a formula.

**Reply by wilkmaciej** [reply]
I was trying to find it, and I'm pretty sure it isn't being built
```
ls -la /opt/homebrew/Cellar/wireshark/3.4.5/bin 
total 3512
drwxr-xr-x  15 wilkmaciej  admin     480 May 12 11:00 .
drwxr-xr-x  14 wilkmaciej  admin     448 May 12 13:12 ..
-r-xr-xr-x   1 wilkmaciej  admin  110048 May 12 11:00 capinfos
-r-xr-xr-x   1 wilkmaciej  admin   88368 May 12 11:00 captype
-r-xr-xr-x   1 wilkmaciej  admin  171072 May 12 11:00 dumpcap
-r-xr-xr-x   1 wilkmaciej  admin  128896 May 12 11:00 editcap
-r-xr-xr-x   1 wilkmaciej  admin  120448 May 12 11:00 idl2wrs
-r-xr-xr-x   1 wilkmaciej  admin   90160 May 12 11:00 mergecap
-r-xr-xr-x   1 wilkmaciej  admin   69392 May 12 11:00 mmdbresolve
-r-xr-xr-x   1 wilkmaciej  admin   90368 May 12 11:00 randpkt
-r-xr-xr-x   1 wilkmaciej  admin  163456 May 12 11:00 rawshark
-r-xr-xr-x   1 wilkmaciej  admin   89808 May 12 11:00 reordercap
-r-xr-xr-x   1 wilkmaciej  admin  211136 May 12 11:00 sharkd
-r-xr-xr-x   1 wilkmaciej  admin  109760 May 12 11:00 text2pcap
-r-xr-xr-x   1 wilkmaciej  admin  333760 May 12 11:00 tshark
```

Looking at cmake args there is ` -DBUILD_wireshark=OFF` [here](https://github.com/Homebrew/homebrew-core/blob/6eff63f9cdd2b5a9969c884e39f0d71583e17ed6/Formula/wireshark.rb#L54)

**Comment by tlk** [comment, marked as answer]
It is likely because [github does not provide M1 VM's](https://github.com/actions/virtual-environments/issues/2187).

Did you try building the application yourself? Clone the project, [install dependencies](https://github.com/wireshark/wireshark/blob/313a6e92740c9588c612ddc03115eab1d4d68ac9/.github/workflows/macos.yml#L27) and [build the thing](https://github.com/wireshark/wireshark/blob/ae750dc141e1658cf87a42eef37e4c755f56a133/README.macos#L72-L75).

**Reply by wilkmaciej** [reply]
I build it myself from wireshark guide. I encountered some errors from cmake about missing files but added `export CMAKE_PREFIX_PATH=/opt/homebrew/Cellar/qt@5/5.15.2/lib/cmake` irrc and I got it working.
